packages/discovery-provider/src/tasks/create_listen_streak_reminder_notifications.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime, timedelta
+
+from sqlalchemy import String, and_, cast, func
+
+from src.models.notifications.notification import Notification
+from src.models.rewards.listen_streak_challenge import ChallengeListenStreak
+from src.tasks.celery_app import celery
+from src.utils.config import shared_config
+from src.utils.structured_logger import StructuredLogger, log_duration
+
+logger = StructuredLogger(__name__)
+env = shared_config["discprov"]["env"]
+
+LISTEN_STREAK_REMINDER = "listen_streak_reminder"
+HOURS_PER_DAY = 24
+LISTEN_STREAK_BUFFER = 6
+LAST_LISTEN_HOURS_AGO = HOURS_PER_DAY - LISTEN_STREAK_BUFFER
+
+
+def get_listen_streak_notification_group_id(user_id, date):
+    return f"{LISTEN_STREAK_REMINDER}:{user_id}:{date}"
+
+
+@log_duration(logger)
+def _create_listen_streak_reminder_notifications(session):
+    now = datetime.now()
+    window_end = now - timedelta(hours=LAST_LISTEN_HOURS_AGO)
+    window_start = now - timedelta(hours=LAST_LISTEN_HOURS_AGO + 1)
```

Comment:
very nitty, but i think `start` coming before `end` is more easy to read

Reply:
im not familiar with celery tasks, qq, may be nothing burgers:

how often does this run?

Say we have 

User listens at 2:00 PM (`last_listen_date = 2:00 PM`)
Current time is 8:00 AM next day (18 hours later)
Perfect 6-hour remaining mark (2:00 PM + 24h = 2:00 PM next day)

But the job runs at 7:30 AM:
*   `window_end` = 7:30 AM - 18h = 1:30 PM (previous day)
*   `window_start` = 1:30 PM - 1h = 12:30 PM (previous day)
This user's 2:00 PM listen won't be detected
Next job run at 8:30 AM would miss it too (`window_end` = 2:30 PM)

is this what would happen with the current setup?